instance/github.py

```diff
@@ -158,13 +159,8 @@ def __init__(self, number, source_fork_name, target_fork_name, branch_name, titl
         self.title = title
         self.username = username
         self.body = body
-
-    @property
-    def extra_settings(self):
-        """
-        Extra settings contained in the PR body
-        """
-        return get_settings_from_pr_body(self.body)
+        self.extra_settings = get_settings_from_pr_body(body)
+        self.extra_settings_dict = yaml.load(self.extra_settings) or {}
```

Comment:
@smarnach 

(a) Yup, they are currently ephemeral; I'm not fully clear on how we'll handle production instances yet, but it seem likely that PR instances will remain this way. We'll likely also manage releases of production instances with PRs, but the production instances themeslves would likely more be linked to a branch - although the rc PR might be the equivalent of stage?

They can however be kept for a certain amount of time (some PRs take months to be merged - the longest almost a year!).

But most importantly - they *can* change. Especially the ansible configuration - the initial PR might not have all the settings, some names might change during reviews. And feature branch PRs might end up being quite complex, and evolving a lot.

(b) Yup, there are always cases where the drawbacks of duplicating data are outweight by other criterias. Having the data on another service is usually a good line - the service might become unavailable, we might want to access the data more often than the service allow us to fetch it, the delays might become too big, etc. But from past experience, the threshold can be fairly high - the cost of keeping duplicated data updated expands exponentially as the application increases in size and complexity. I've found it useful to prioritize having getting the right data structure, over things like having efficient code for example - human brain cycles are much more expensive than computer ones : )

> I'd like to remind you that the last bug I fixed was caused by too many things being properties and recomputed everytime when actually they shouldn't be, so you have to look at each individual case to figure out the right approach.

Yup, definitely not an absolute, just something I'd like to be mindful of.